docs/sdk/src/reference_docs/cli.rs

```diff
@@ -1,7 +1,105 @@
-//! # Command Line Arguments
+//! # Substrate CLI
 //!
+//! Let's see some examples of typical CLI arguments used when setting up and running a
+//! Substrate-based blockchain. We use the [`substrate-node-template`](https://github.com/substrate-developer-hub/substrate-node-template)
+//! on these examples.
 //!
-//! Notes:
+//! #### Checking the available CLI arguments
+//! ```bash
+//! ./target/debug/node-template --help
+//! ```
+//! - `--help`: Displays the available CLI arguments.
 //!
-//! - Command line arguments of a typical substrate based chain, and how to find and learn them.
-//! - How to extend them with your custom stuff.
+//! #### Starting a Local Substrate Node in Development Mode
+//! ```bash
+//! ./target/release/node-template \
+//! --dev
+//! ```
+//! - `--dev`: Runs the node in development mode, using a pre-defined development chain
+//!   specification.
+//! This mode ensures a fresh state by deleting existing data on restart.
+//!
+//! #### Generating Custom Chain Specification
+//! ```bash
+//! ./target/debug/node-template \
+//! build-spec \
+//! --disable-default-bootnode \
+//! --chain local \
+//! > customSpec.json
+//! ```
+//!
+//! - `build-spec`: A subcommand to generate a chain specification file.
+//! - `--disable-default-bootnode`: Disables the default bootnodes in the node template.
+//! - `--chain local`: Indicates the chain specification is for a local development chain.
+//! - `> customSpec.json`: Redirects the output into a customSpec.json file.
+//!
+//! #### Converting Chain Specification to Raw Format
+//! ```bash
+//! ./target/debug/node-template build-spec \
+//! --chain=customSpec.json \
+//! --raw \
+//! --disable-default-bootnode \
+//! > customSpecRaw.json
+//! ```
+//!
+//! - `--chain=customSpec.json`: Uses the custom chain specification as input.
+//! - `--disable-default-bootnode`: Disables the default bootnodes in the node template.
+//! - `--raw`: Converts the chain specification into a raw format with encoded storage keys.
+//! - `> customSpecRaw.json`: Outputs to customSpecRaw.json.
+//!
+//! #### Starting the First Node (Bootnode) in a Private Network
```

Comment:
```suggestion
//! #### Starting the First Node in a Private Network
```

If you want to mention boot node, I would do it somewhere below in a sentence